src/mlpack/methods/naive_bayes/nbc_main.cpp

```diff
@@ -188,14 +191,17 @@ static void mlpackMain()
     model->nbc.Classify(testingData, predictions, probabilities);
     Timer::Stop("nbc_testing");
 
-    if (CLI::HasParam("output"))
+    if (CLI::HasParam("output") || CLI::HasParam("predictions"))
     {
       // Un-normalize labels to prepare output.
       Row<size_t> rawResults;
       data::RevertLabels(predictions, model->mappings, rawResults);
 
-      // Output results.
-      CLI::GetParam<Row<size_t>>("output") = std::move(rawResults);
+      CLI::GetParam<Row<size_t>>("predictions") = std::move(rawResults);
+      if (CLI::HasParam("output"))
+      {
+         CLI::GetParam<Row<size_t>>("output") = std::move(predictions);
+      }
```

Comment:
Why are you setting one to `predictions` and one to `rawResults`?  I think both should be set to `rawResults`.

Reply:
On running the tests if we set both to rawResults tests would fail whereas this way they will pass and algorithm will run correctly too

Reply:
@rcurtin 
Perhaps,i could do this too
` ` `
predictions=rawResults;
if(CLI::HasParam("output"))
{
    CLI::GetParam<Row<size_t>>("output") = std::move(predictions);
 }
` ` `
this will work too

Reply:
The algorithm does not run correctly if you set one to `rawResults` and one to `predictions` because those both contain different values.

The updated code still has a problem.  The `if` statement outside this block assumes that either `CLI::HasParam("output")` is true or `CLI::HasParam("predictions")` is true.  But then inside, it assumes that `predictions` has been specified and calls `GetParam()` on it (even if it has not been specified).  Can you clean this up?

Also, the line `predictions = rawResults` makes the entire code wrong.  Did you read the documentation for `data::RevertLabels()` to understand what it does and why that line is there?  If you do, you will understand why `predictions` and `rawResults` are different and should be treated as such.

Reply:
> The updated code still has a problem. The if statement outside this block assumes that either CLI::HasParam("output") is true or CLI::HasParam("predictions") is true. But then inside, it assumes that predictions has been specified and calls GetParam() on it (even if it has not been specified). Can you clean this up?

This problem still exists in the code, can you fix it please?